File: replace_method.py
#!/usr/bin/env python3
import re

# Read files
with open('src/services/ultraDetailedIngestionAgent.js', 'r') as f:
    original = f.read()

with open('/tmp/garment-fix-inner/garment-classification-fix/ultraDetailedIngestionAgent-UPDATED.js', 'r') as f:
    updated_file = f.read()

# Extract the new method from the updated file (skip comment header)
lines = updated_file.split('\n')
method_start = None
for i, line in enumerate(lines):
    if line.strip() == 'getComprehensivePrompt() {':
        method_start = i
        break

if method_start is None:
    print("ERROR: Could not find method in updated file")
    exit(1)

new_method_lines = lines[method_start:]
new_method = '\n'.join(new_method_lines)

# Find and replace the old method in original
# Pattern: find getComprehensivePrompt() { ... matching closing brace }
pattern = r'(\s+)getComprehensivePrompt\(\) \{.*?^\s+\}'
replacement = r'\1getComprehensivePrompt() {\n' + '\n'.join(['  ' + line for line in new_method_lines[1:]]) + '\n  }'

# Use DOTALL to match across newlines, MULTILINE for ^ to match line starts
result = re.sub(pattern, replacement, original, count=1, flags=re.DOTALL | re.MULTILINE)

if result == original:
    print("ERROR: No replacement made - pattern didn't match")
    exit(1)

# Write back
with open('src/services/ultraDetailedIngestionAgent.js', 'w') as f:
    f.write(result)

print("✅ Successfully replaced getComprehensivePrompt() method!")
print(f"   New method: {len(new_method_lines)} lines")

File: replace-method.js
const fs = require('fs');

// Read the files
const originalFile = 'src/services/ultraDetailedIngestionAgent.js';
const originalContent = fs.readFileSync(originalFile, 'utf8');
const newMethod = fs.readFileSync('/tmp/new-method.txt', 'utf8');

// Find the method boundaries in the original file
const lines = originalContent.split('\n');
let startLine = -1;
let endLine = -1;

for (let i = 0; i < lines.length; i++) {
  if (lines[i].trim() === 'getComprehensivePrompt() {') {
    startLine = i;
    // Find the matching closing brace
    let braceCount = 0;
    let foundOpenBrace = false;
    
    for (let j = i; j < lines.length; j++) {
      for (let char of lines[j]) {
        if (char === '{') {
          braceCount++;
          foundOpenBrace = true;
        }
        if (char === '}') braceCount--;
      }
      
      if (foundOpenBrace && braceCount === 0) {
        endLine = j;
        break;
      }
    }
    break;
  }
}

if (startLine === -1 || endLine === -1) {
  console.error('Could not find method boundaries');
  console.log('Start:', startLine, 'End:', endLine);
  process.exit(1);
}

console.log(`Found method: lines ${startLine + 1} to ${endLine + 1}`);

// Build the new content
const before = lines.slice(0, startLine).join('\n');
const after = lines.slice(endLine + 1).join('\n');
const newContent = before + '\n' + newMethod.trim() + '\n' + after;

// Write it back
fs.writeFileSync(originalFile, newContent, 'utf8');

console.log('✅ Method replaced successfully!');
console.log(`   Old: ${endLine - startLine + 1} lines`);
console.log(`   New: ${newMethod.trim().split('\n').length} lines`);

